addons/web/static/tests/views/form/form_view_tests.js

```diff
@@ -4046,6 +4046,68 @@ QUnit.module("Views", (hooks) => {
         await editInput(target, ".o_field_widget[name=foo] input", "tralala");
     });
 
+    QUnit.test("onchange send relation parent field values (including readonly)", async function (assert) {
+        assert.expect(1);
+
+        serverData.models.user.fields.login = { string: "Login", type: "char" };
+        serverData.models.user.onchanges = {};
+        serverData.models.user.onchanges.name = (obj) => {
+            // like computed field that depends on "name" field
+            obj.login = obj.name.toLowerCase() + '@example.org'
+        }
+        serverData.models.partner.onchanges.qux = () => {};
+
+        let checkOnchange = false;
+        await makeView({
+            type: "form",
+            resModel: "user",
+            serverData,
+            arch: `
+                <form>
+                    <field name="name"/>
+                    <field name="login" readonly="True"/>
+                    <field name="partner_ids">
+                        <tree editable="top">
+                            <field name="qux"/>
+                        </tree>
+                    </field>
+                </form>`,
+            resId: 17,
+            mockRPC(route, args) {
+                if (args.method === "onchange" && checkOnchange) {
+                    assert.deepEqual(
+                        args.args[1],
+                        {
+                            qux: 12.4,
+                            user_id: {
+                                id: 17,
+                                name: "Test",
+                                login: "test@example.org",
+                                partner_ids: [
+                                    [
+                                        0,
+                                        args.args[1].user_id.partner_ids[0][1],
+                                        { qux: 0 },
+                                    ],
+                                ],
+                            }
+                        },
+                        "should send the values (including readonly) for the relational parent field"
+                    );
+                }
+            },
+        });
+        // trigger an onchange that update a readonly field by modifying user name
+        await editInput(target, '.o_field_widget[name=name] input', 'Test')
+
+        // add a o2m row
+        await click(target.querySelector(".o_field_x2many_list_row_add a"));
+
+        // trigger an onchange by modifying foo
+        checkOnchange = true;
+        await editInput(target, ".o_field_one2many .o_field_widget[name=qux] input", "12.4");
+    })
```

Comment:
could you lint/prettify your code? You can enable the tooling via ./addons/web/tooling/enable.sh (it's a pre-commit hook)

Reply:
done, thanks for the review :)